segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -838,6 +839,12 @@ public void readTableEntriesDelta(WireCommands.ReadTableEntriesDelta readTableEn
         val timer = new Timer();
         val result = new DeltaIteratorResult<BufferView, Map.Entry<WireCommands.TableKey, WireCommands.TableValue>, DeltaIteratorState>(
                 segment.getBytes().length + WireCommands.TableEntriesRead.HEADER_BYTES);
+        // Guards against two cases: an empty segment, or a starting position that exceeds the segment length.
+        segmentStore.getStreamSegmentInfo(segment, TIMEOUT)
+                .thenAccept(properties -> {
+                    result.setState(TableEntryDeltaIterator.initialState(properties, fromPosition));
```

Comment:
I'm not quite sure on the performance impact (if any) but certain properties about the Segment are needed to define some `DeltaIteratorState` instance. It did not seem feasible to be able to use the information acquired in `ContainerTableExtensionImpl.newDeltaIterator` as that would require several changes about the semantics of the iteration.